### PR TITLE
Simplify command args parsing

### PR DESCRIPTION
**Pull request by josefkarasek**
## Simplify command args parsing

Command arguments are now parsed in multiple different places.
This PR merges existing arg parsing code in one place. The same with unit tests.

## Related issue(s)
#703

### CODE REVIEW

File: pkg/execute/config_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func TestConfigExecutorShowConfig(t *testing.T) {
 				Conversation:   Conversation{Alias: channelAlias, ID: "conv-id"},
 				Platform:       config.SlackCommPlatformIntegration,
 				ClusterName:    configTestClusterName,
-				ExecutorFilter: newExecutorTextFilter("", "config"),
```
Comment (josefkarasek) [inline pkg/execute/config_test.go:33]: `Command string` is now in `CommandContext struct`

File: pkg/execute/executor.go [hunk 2]
```diff
@@ -162,27 +162,31 @@ func (e *DefaultExecutor) Execute(ctx context.Context) interactive.Message {
...
+		CommGroupName:   e.commGroupName,
+		BotName:         e.notifierHandler.BotName(),
 		User:            e.user,
 		Conversation:    e.conversation,
 		Platform:        e.platform,
 		NotifierHandler: e.notifierHandler,
 	}
-	execFilter, err := extractExecutorFilter(rawCmd)
+
+	params, err := params.RemoveBotkubeRelatedFlags(rawCmd)
```
Comment (josefkarasek) [inline pkg/execute/executor.go:177]: Would something like `ParseAndRemoveBotkubeFlags` sound better?

File: pkg/execute/executor.go [hunk 3]
```diff
@@ -218,7 +222,7 @@ func (e *DefaultExecutor) Execute(ctx context.Context) interactive.Message {
 			e.log.Errorf("while executing kubectl: %s", err.Error())
 			return empty
 		}
-		return respond(execFilter.Apply(out), cmdCtx)
```
Comment (josefkarasek) [inline pkg/execute/executor.go:221]: `Apply()` is called in `func respond`

File: pkg/execute/executor.go [hunk 4]
```diff
@@ -149,6 +147,8 @@ type CommandContext struct {
 	CommGroupName   string
 	BotName         string
 	RawCmd          string
+	CleanCmd        string
+	WithClusterName string
```
Comment (pkosiec) [inline pkg/execute/executor.go:151]: Let's rename that to `ProvidedClusterName`
```suggestion
	ProvidedClusterName string
```

it'll be more verbose. WDYT?

File: pkg/execute/executor_filter_test.go [hunk 5]
```diff
@@ -53,137 +53,10 @@ daemonset.apps/kindnet      3         3         3       3            3
...
-	var txFilter executorFilter = newExecutorTextFilter("kind", "")
+	var txFilter executorFilter = newExecutorTextFilter("kind")
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			assert.Equal(t, tc.expected, txFilter.Apply(tc.text))
 		})
 	}
 }
-
-func TestExtractExecutorFilter_NoErrors(t *testing.T) {
```
Comment (pkosiec) [inline pkg/execute/executor_filter_test.go:64]: I can't find these tests. Were they removed?

Comment (josefkarasek) [reply pkg/execute/executor_filter_test.go:64]: Good point. I didn't notice they touched `filter.Apply()`.

File: pkg/execute/executor.go [hunk 6]
```diff
@@ -83,33 +76,30 @@ func (e *DefaultExecutor) Execute(ctx context.Context) interactive.Message {
...
 	}
-	cmdCtx.ExecutorFilter = execFilter
 
-	args, err := shellwords.Parse(strings.TrimSpace(execFilter.FilteredCommand()))
-	if err != nil {
-		e.log.Errorf("while parsing command %q: %s", execFilter.FilteredCommand(), err.Error())
-		return respond("Cannot parse command. Please use 'help' to see supported commands.", cmdCtx)
-	}
+	cmdCtx.CleanCmd = flags.CleanCmd
+	cmdCtx.ClusterName = flags.ClusterName
```
Comment (pkosiec) [inline pkg/execute/executor.go:98]: I spotted a bug:

```suggestion
	cmdCtx.ProvidedClusterName = flags.ClusterName
```

Correct? Please test your PR by yourself before pinging me next time as I'd like to finally test an accept it 🙂

File: pkg/execute/params.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+	filterFlagParseErrorMsg = `incorrect use of --filter flag: could not parse flag in %s
+error: %s
+Use --filter="value" or --filter value`
+)
+
+var (
+	clusterNameFlagRegex = regexp.MustCompile(`--cluster-name[=|\s]*(\S*)`)
+)
+
+// OptionalParams contains cmd line arguments for executors
```
Comment (pkosiec) [inline pkg/execute/params.go:28]: Please add `.` after all comments in this file. Thanks!

File: pkg/execute/params.go [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+type OptionalParams struct {
+	CleanCmd     string
+	Filter       string
+	ClusterName  string
+	TokenizedCmd []string
+}
```
Comment (pkosiec) [inline pkg/execute/params.go:34]: BTW consider renaming this file to `flags` and renaming this struct with e.g. simply Flags?

```suggestion
type Flags struct {
	CleanCmd     string
	Filter       string
	ClusterName  string
	TokenizedCmd []string
}
```

File: pkg/execute/params.go [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+// ParseBotkubeFlags parses raw cmd and removes optional params with flags
+func ParseBotkubeFlags(cmd string) (OptionalParams, error) {
```
Comment (pkosiec) [inline pkg/execute/params.go:37]: Why `Botkube` in this name? Please avoid using the app name in functions / methods. If you'd like to highlight they are global, maybe `ParseGlobalFlags` will be better? What do you think?

**Review by pkosiec — COMMENTED** [review]
Almost there! I spotted one small bug which blcoked me from testing this PR. Please fix it, thanks!

**Review by pkosiec — APPROVED** [review]
Tested, works well 👌 
tenor-266226541

**Pull request merged**